src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -171,11 +193,99 @@ private string GetMethodName(ISyntaxFactsService syntaxFacts, SyntaxNode express
             return argsArray.ElementType.IsReferenceType;
         }
 
-        internal abstract bool TryGetArgsArgumentType(
+        private bool TryGetArgsArgumentType(
             SemanticModel semanticModel,
             SeparatedSyntaxList<SyntaxNode> arguments,
             ImmutableArray<IParameterSymbol> parameters,
-            out ITypeSymbol argsArgumentType);
+            out ITypeSymbol argsArgumentType)
+        {
+            var nameOfArgsParameter = "args";
+            if (!TryGetArgument(
+                semanticModel, 
+                nameOfArgsParameter, 
+                arguments, 
+                parameters, 
+                out var argsArgument))
+            {
+                argsArgumentType = null;
+                return false;
+            }
+
+            argsArgumentType = GetArgumentExpressionType(semanticModel, argsArgument);
+            return true;
+        }
+
+        protected bool TryGetArgument(
+            SemanticModel semanticModel,
+            string searchArgumentName,
+            SeparatedSyntaxList<SyntaxNode> arguments,
+            ImmutableArray<IParameterSymbol> parameters,
+            out SyntaxNode argumentSyntax)
+        {
+            argumentSyntax = null;
+
+            // First, look for a named argument that matches
+            var namedArguments = GetMatchingNamedArgument(arguments, searchArgumentName);
+
+            if (namedArguments != null)
+            {
+                argumentSyntax = namedArguments;
+                return true;
+            }
+
+            // If no named argument exists, look for the named parameter
+            // and return the corresponding argument
+            var namedParameters = parameters.Where(p => p.Name.Equals(searchArgumentName));
+
+            if (namedParameters.Count() != 1)
+            {
+                return false;
+            }
+
+            var namedParameter = namedParameters.Single();
```

Comment:
PERF: Again, consider, not using `Where()` + `Count()` + `Single()` in a hot path in an analyzer.